### DISCUSSION TITLE
how to load and div by its ID to generate pdf ... load_html(#dom object)

### DISCUSSION THREAD

**Discussion by adityagade** [dompdf/dompdf#3040] [Q&A]
Hello,

im using dompdf to convert html to pdf but my requirement is to load existing div or an table using its ID or CLASS NAME to generate the pdf.

for example 
instead of using following:

>`$dompdf->load_html("<div id="mydiv">Hello world</div>");`

want to use following:

> `$dompdf->load_html("#mydiv");`

**Comment by bsweeney** [comment]
Dompdf only takes straight HTML so you'll have to pre-parse your document to get the desired value. [As outlined on StackOverflow](https://stackoverflow.com/a/11226049/264628), something like:

```
$doc = new DOMDocument();
$doc->loadHTML($html);
$element = $dom->getElementById("mydiv");
$dompdf->loadHtml($element->C14N());
```
